_posts/2021-08-20-gsoc-project-summary.md

```diff
@@ -0,0 +1,55 @@
+
+Now the world generation process works quite differently. It begins with `GalaxyBuilder` (previously referred to as `WorldBuilder`). This class uses reflection to find all available `SolarSystemGenerator` classes. A `Generator` class represents a feature of the Destination: Sol world, such as a Planet, Maze, Asteroid Belt, or even a SolarSystem. The `GalaxyBuilder` loads `Generators` that represent entire SolarSystems. It then finds a position for each SolarSystem (based on the SolarSystem size) and then tells each `SolarSystemGenerator` to run its build process.
+
+This build process has two sections. First, properties related to this specific `SolarSystemGenerator`, such as its name or its config file, are set. Then, a method is called to initiate the build process of each Feature (Planet, Maze, or Asteroid Belt) that will populate this SolarSystem. Then, once each of those build processes complete, the `SolarSystemGenerator` creates a `SolarSystem` object for the game's runtime code to use. 
+
+We mentioned that each of those `SolarSystemGenerator` build processes initiates `FeatureGenerator` build processes. These work very similarly to the process for `SolarSystemGenerators`. However, they only contain the step of setting the details specific to that feature. For example, within the `build()` method of a `PlanetGenerator` implementation, the code will set the Planet config file, ground height, atmosphere height, orbit speed, rotation speed, and name. Once the values are set, the `build()` method concludes by instantiating a `Planet` object for the game to use.
+
+Each `Generator` class, whether a `SolarSystemGenerator` or `FeatureGenerator`, is an implementation of an abstract class that requires the implementation to have a `build()` method. These abstract classes also provide all the relevant methods for determining the details of the particular object you are generating. 
+
+# Flow of Galaxy Generation
+![World Generation Sequence](/img/posts/wrap-up-2021/FlowOfGeneration.png)
+
+I created a class called `Orbital` to help keep track of where Features should go within a SolarSystem. You can think of the `Orbitals` as concentric rings that surround the SolarSystem center. They each have a certain width and begin a certain distance from the center of the System. Depending on its size, a SolarSystem will have 3, 5, or 7 `Orbitals`. There is also an additional `Orbital` for Mazes, which are always on the edge of the SolarSystem.
```

Comment:
Another tiny nitpick: I'd say that an "orbital" is simply something that orbits _something_ not necessarily a star. A star gate thingie for instance is sort of an orbital for its parent planet. This may be a moot point right now though as I figure only solar systems have orbital _layers_ - although IIRC planets do have both high and low orbiting enemies, then also the star gate layer above.